tests/testthat/test-social-smells.R

```diff
@@ -0,0 +1,112 @@
+test_that("Using Perceval version other than 0.12.24 will fail for parse_mbox", {
+  perceval_path <- "../../../tools/bin/perceval-incompatible-version"
+  mbox_path <- "../../../../rawdata/mbox/project.mbox"
+  expect_error(parse_mbox(perceval_path, mbox_path))
+})
+test_that("Correct config file parameters and Perceval version 0.12.24 work", {
+  perceval_path <- "../../../tools/bin/perceval"
+  mbox_path <- "../../../../rawdata/mbox/project.mbox"
+  jira_issue_comments_path <- "../../../../rawdata/issue_tracker/project_issue_comments.json"
+  expect_no_error(parse_mbox(perceval_path, mbox_path))
+  expect_no_error(parse_jira(jira_issue_comments_path))
+})
+test_that("Incorrect paths to files passed to parse functions fails", {
+  perceval_path <- "../../../tools/bin/perceval"
+  perceval_path_bad <- "/bad/perceval/path"
+  mbox_path <- "../../../../rawdata/mbox/project.mbox"
+  mbox_path_bad <- "/bad/mbox/path"
+  jira_issue_comments_path_bad <- "/bad/jira/path"
+  expect_error(parse_mbox(perceval_path_bad, mbox_path))
+  expect_error(parse_mbox(perceval_path, mbox_path_bad))
+  expect_warning(expect_error(parse_jira(jira_issue_comments_path_bad)))
+})
+test_that("Empty file path passed to parse functions fails", {
+  perceval_path <- "../../../tools/bin/perceval"
+  mbox_path <- "../../../../rawdata/mbox/project.mbox"
+  # If a user forgot to set a config parameter, they would have something like
+  # the variable below
+  empty_path <-
+  # What happens if config parameter(s) left blank
+  expect_error(parse_mbox(perceval_path, empty_path))
+  expect_error(parse_mbox(empty_path, perceval_path))
+  expect_error(parse_mbox(empty_path, empty_path))
+  expect_error(parse_jira(empty_path))
+})
+test_that("Absolute filepaths passed to parsers work", {
+  perceval_path_rel <- "../../../tools/bin/perceval"
+  mbox_path_rel <- "../../../../rawdata/mbox/project.mbox"
+  perceval_path_abs <- "/Users/absolute-path/tools/bin/perceval"
+  mbox_path_abs <- "/Users/absolute-path/rawdata/mbox/project.mbox"
+  jira_path_abs <- "/Users/absolute-path/rawdata/issue_tracker/project_issue_comments.json"
+  expect_no_error(parse_mbox(perceval_path_rel, mbox_path_abs))
+  expect_no_error(parse_mbox(perceval_path_abs, mbox_path_rel))
+  expect_no_error(parse_mbox(perceval_path_abs, mbox_path_abs))
+  expect_no_error(parse_jira(jira_path_abs))
+})
+test_that("Parsers produce tables with correct columns and expected size", {
```

Comment:
Hi Carlos,
shouldn't we expect the user to have a configuration file set up as well with all the parameters for social_smells_showcase.Rmd? Also, do you have a preference for which config file the user is using in these tests (thrift, helix, geronimo, etc)?

Also, in regards to the mini git, could we add a test data file in the testthat directory that contains several commits from the kaiaulu git log and use that for the test? I was thinking of following this convention https://r-pkgs.org/testing-design.html#storing-test-data where there's a fixture folder with the test data.

Reply:
We can assume what is on Kaiaulu repo. For example, a config file we host. The tools.yml we have also has to be modified to comply to your new unit tests. We can then test if the user, in modifying the configs to their needs, are not violating the assumptions we set forth in the repo. But since we do not have the data hosted, we can't assume the data.

No preference for the choice of config file, so long it is already in the repo. 

I am not sure if we want to start hosting test data, because I am not confident this will scale. Kaiaulu is a tool that has interoperability to a lot of different data sources. It implements some features of its own, but at its most basic purpose, it gives a common model to reason and combines all of them. If we start hosting test data for every single interface, our repo will clog, and I am not sure the folks over CRAN will be too happy either whenever Kaiaulu is made into an official R package. 

I'd rather we create the sample data on the fly instead. This will also give us more control over what we wish to test. For example, if we extend the git log sample creation function, to parameterize the number of files, and number of commits, we can now create very precise unit tests to see how parse_gitlog() behave. Meanwhile, using any sample git log, like Kaiaulu's, we may be missing out on these scenarios. Likewise to different conventions of annotating issue id on commits, and the body, which are parsed to construct issue networks, etc. 

I think hosting sample data has its uses, for example, if we were testing a machine learning algorithm which has randomize initialization, off the same data source over and over again. But I may reconsider this in the future.